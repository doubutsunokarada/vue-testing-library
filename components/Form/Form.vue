<template>
  <h1>アカウント情報</h1>
  <form @submit.prevent="onSubmit">
    <h2>アカウント名:</h2>
    <p>{{ name }}</p>
    <div><button type="submit">編集</button></div>
  </form>
</template>

<script setup lang="ts">
import { ref } from "vue";

interface Props {
  name?: string;
  onSubmit?: () => void;
}

const props = defineProps<Props>();

const name = ref(props.name ?? "Bob");
const onSubmit = props.onSubmit ?? (() => {});
</script>
