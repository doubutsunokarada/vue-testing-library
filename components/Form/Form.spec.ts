import { render, screen } from "@testing-library/vue";
import { userEvent } from "@testing-library/user-event";
import Form from "./Form.vue";
import { expect, test, vi } from "vitest";

test("名前の表示", () => {
  render(Form, { props: { name: "Taro" } });
  expect(screen.getByText("Taro")).toBeInTheDocument();
});

test("見出しの表示", () => {
  render(Form);
  expect(screen.getByRole("heading", { level: 1 })).toBeInTheDocument();
  expect(screen.getByRole("heading", { level: 2 })).toBeInTheDocument();
});

test("buttonの表示", () => {
  render(Form);
  expect(screen.getByRole("button")).toBeInTheDocument();
});

test("見出しのテキスト", () => {
  render(Form);
  expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent(
    "アカウント情報"
  );
  expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent(
    "アカウント名:"
  );
});

test("onSubmitの呼び出し", async () => {
  const fn = vi.fn();
  const user = userEvent.setup();
  render(Form, { props: { name: "", onSubmit: fn } });
  await user.click(screen.getByRole("button"));
  expect(fn).toHaveBeenCalled();
});

test("Taroの表示", () => {
  // const { container } = render(Form, { props: { name: "Taro" } });
  // Regression
  const { container } = render(Form, { props: { name: "Hoge" } });
  expect(container).toMatchSnapshot();
});
