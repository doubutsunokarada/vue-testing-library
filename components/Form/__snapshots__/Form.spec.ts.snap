// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Taroの表示 1`] = `
<div>
  
  <h1>
    アカウント情報
  </h1>
  <form>
    <h2>
      アカウント名:
    </h2>
    <p>
      Hoge
    </p>
    <div>
      <button
        type="submit"
      >
        編集
      </button>
    </div>
  </form>
  
</div>
`;
